arctic/store/version_store.py

```diff
@@ -301,12 +303,21 @@ def _read_handler(self, version, symbol):
             handler = self._bson_handler
         return handler
 
+    @staticmethod
+    def handler_can_write_type(handler, data):
+        type_method = getattr(handler, "can_write_type", None)
+        if callable(type_method):
+            return type_method(data)
+        return False
+
     def _write_handler(self, version, symbol, data, **kwargs):
         handler = None
         for h in _TYPE_HANDLERS:
             if h.can_write(version, symbol, data, **kwargs):
                 handler = h
                 break
+            if AVOID_FALLBACK_HANDLERS and self.handler_can_write_type(h, data):
+                raise ArcticException( "Not falling back to default handler for %s" % symbol )
```

Comment:
This behaviour is slightly different than the exception message suggests. If I have multiple handlers for one type this new code will prevent ones further down the list being used as well as preventing the default handler being used.